### PR TITLE
evm: Add executor logs to execute_evm response

### PR DESCRIPTION
**Pull request by tgmichel**
Rel https://github.com/paritytech/frontier/pull/103

Return executor logs to enable Transaction Receipt Log event subscriptions in Frontier.

### CODE REVIEW

**Comment by tgmichel** [comment]
@sorpaas please when you have a minute

File: frame/evm/src/lib.rs [hunk 1]
```diff
@@ -629,11 +629,20 @@ impl<T: Trait> Module<T> {
 		);
 		executor.deposit(source, total_fee.saturating_sub(actual_fee));
 
+		let mut logs_result: Vec<Log> = Vec::new();
 		if apply_state {
 			let (values, logs) = executor.deconstruct();
-			backend.apply(values, logs, true);
+			let logs_data = logs.into_iter().map(|x| x ).collect::<Vec<_>>().clone();
```
Comment (sorpaas) [inline frame/evm/src/lib.rs:635]: What's the reason that we skip handling logs return data when `apply_state` is false?

Comment (tgmichel) [reply frame/evm/src/lib.rs:635]: No reason at all, just a matter if we want to return the log data on gas estimation or not. Do you think we should?

Comment (sorpaas) [reply frame/evm/src/lib.rs:635]: Yes. I think if we're returning logs, we should always make sure the value is valid.

**Comment by bkchr** [comment]
Any update?

**Comment by tgmichel** [comment]
Yes, my bad. @sorpaas If there is nothing more to add than returning logs on estimate gas, can we approve this and get it through? I will be back to it (and the pub/sub) soon and add that if we require it!

**Comment by sorpaas** [comment]
@tgmichel Would you mind to change that line to always push to log result regardless of `apply_state`? My main worry is that if the caller misuses the API, expect to get logs but receive an empty array.

**Comment by tgmichel** [comment]
@sorpaas done

**Review by sorpaas — APPROVED** [review]

**Comment by tgmichel** [comment]
@sorpaas failing on `gitlab-test-linux-stable`. What is this related to? Seemed to pass in the previous check.

**Review by kianenigma — APPROVED** [review]
@sorpaas Given strict merge rules, and these PRs not directly affecting polkadot/kusama, I guess you should make a project for them and make yourself the lead to facilitate merges.

**Comment by sorpaas** [comment]
bot merge

**Pull request merged**